staging/src/k8s.io/apiserver/pkg/authentication/authenticator/metrics_test.go

```diff
@@ -0,0 +1,175 @@
+				`,
+		},
+		{
+			desc: "auth failed with error",
+			request: WrapInInstrumentedRequest(
+				RequestFunc(
+					func(_ *http.Request) (*Response, bool, error) {
+						return nil, false, &InstrumentedError{
+							AuthenticatorName: "instrumented-auth",
+							error:             errors.New("some error"),
+						}
+					}),
+				"instrumented-auth"),
+			want: `
+		    # HELP authentication_attempts [ALPHA] Counter of authenticated attempts.
+		    # TYPE authentication_attempts counter
+		    authentication_attempts{authenticator="instrumented-auth", result="error",} 1
+				`,
+		},
+		{
+			desc: "auth failed with status false",
+			request: WrapInInstrumentedRequest(
+				RequestFunc(
+					func(_ *http.Request) (*Response, bool, error) {
+						return &Response{User: &user.DefaultInfo{Name: "admin"}}, false, nil
+					}),
+				"instrumented-auth"),
+			want: `
+		    # HELP authentication_attempts [ALPHA] Counter of authenticated attempts.
+		    # TYPE authentication_attempts counter
+		    authentication_attempts{authenticator="instrumented-auth", result="failure"} 1
+				`,
+		},
+		{
+			desc: "auth failed due to audiences not intersecting",
+			request: WrapInInstrumentedRequest(
+				RequestFunc(
+					func(_ *http.Request) (*Response, bool, error) {
+						return &Response{
+								User:      &user.DefaultInfo{Name: "admin"},
+								Audiences: Audiences{"audience-x"}},
+							true,
+							nil
+					}),
+				"instrumented-auth"),
+			apiAudience: Audiences{"audience-y"},
+			want: `
+		    # HELP authentication_attempts [ALPHA] Counter of authenticated attempts.
+		    # TYPE authentication_attempts counter
+		    authentication_attempts{authenticator="instrumented-auth", result="error"} 1
+				`,
+		},
+		{
+			desc: "audiences not supplied in the response",
+			request: WrapInInstrumentedRequest(
+				RequestFunc(
+					func(_ *http.Request) (*Response, bool, error) {
+						return &Response{
+								User: &user.DefaultInfo{Name: "admin"},
+							},
+							true,
+							nil
+					}),
+				"instrumented-auth"),
+			apiAudience: Audiences{"audience-y"},
+			want: `
+		    # HELP authenticated_user_requests [ALPHA] Counter of authenticated requests broken out by username.
+				# TYPE authenticated_user_requests counter
+				authenticated_user_requests{authenticator="instrumented-auth", username="admin"} 1
+		    # HELP authentication_attempts [ALPHA] Counter of authenticated attempts.
+		    # TYPE authentication_attempts counter
+		    authentication_attempts{authenticator="instrumented-auth", result="success"} 1
+				`,
+		},
+		{
+			desc: "audiences not supplied to the handler",
+			request: WrapInInstrumentedRequest(
+				RequestFunc(
+					func(_ *http.Request) (*Response, bool, error) {
+						return &Response{
+								User:      &user.DefaultInfo{Name: "admin"},
+								Audiences: Audiences{"audience-x"},
+							},
+							true,
+							nil
+					}),
+				"instrumented-auth"),
```

Comment:
Can you add a couple tests using the union authenticator?

Reply:
Union tests are in the filter/metrics_test, see comment below on why I split them out.